Makefile

```
@@ -42,12 +43,17 @@ rapidwright:
 	cd third_party/RapidWright && make
 
```

This is what I'm thinking for the submodule target:
```suggestion

submodules:
	git submodule init
	git submodule update

```
and then add this as a dependency for all the third party install mechanisms.

---

I tried an alternate approach where instead I defined a function that would do this same thing since I wasn't able to get submodule targets working like you're looking for.

---

The issue with this approach is you will run `git submodule init`  and `git submodule update` once for each install target.  I'll make some suggested changes that should do the trick.